client/scripts/views/components/topic_selector.ts

```diff
@@ -17,19 +14,15 @@ const TopicSelector: m.Component<{
   error: string;
   selectedTopic?: OffchainTopic;
 }> = {
-  oninit: (vnode) => {
-    const { defaultTopic, topics } = vnode.attrs;
+  view: (vnode) => {
+    const { defaultTopic, featuredTopics, tabindex, topics, updateFormData } = vnode.attrs;
     if (defaultTopic === false) {
       vnode.state.selectedTopic = undefined;
     } else if (defaultTopic && typeof defaultTopic === 'string') {
       vnode.state.selectedTopic = topics.find((t) => t.name === defaultTopic);
     } else if (defaultTopic && defaultTopic instanceof OffchainTopic) {
       vnode.state.selectedTopic = defaultTopic;
     }
```

Comment:
Will this overwrite selectedTopic every time the view is rendered?

Reply:
It does, but it stays in sync with defaultTopic via the ThreadForm, and it ends up simpler than implementing and sharing a switch (e.g. `alreadySet`) across the components. I've tried just about every operation I could think on setting/resetting/refreshing topics, and carefully logged out the selectedTopic—it's behaved as it should.

Reply:
Okay, if so, then we should remove selectedTopic from vnode.state and instantiate the variable from scratch every time at the start of the `view` function.

Reply:
Yes... I think because it gets updated via several event listeners, I was worried it wouldn't stay up-to-date, but QAing now it seems to do fine with a plain old variable. Should be set.